_posts/en/2019-10-21-schnorr-taproot-workshop.md

```diff
@@ -0,0 +1,189 @@
+{:.center}
+
+James Chiang gives an overview of the bip-taproot/bip-tapscript proposals. This
+notebook demonstrates how we'll create transaction outputs, then spend them and
+verify that the spend is valid.
+
+#### 2.1 Segwit Version 1
+
+[![Segwit Version 1](/img/posts/taproot-workshop/segwit-version-1.png)](https://www.youtube.com/watch?v=n-jAUaSkcAA&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=6)
+{:.center}
+
+James shows how to create segwit version 1 transaction outputs and spend them
+using the key path spend.
+
+#### 2.2 Taptweak
+
+[![Taptweak](/img/posts/taproot-workshop/taptweak.png)](https://www.youtube.com/watch?v=EkGbPxAExdQ&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=7)
+{:.center}
+
+James explains what a key tweak is, and how a tweak can be
+used to commit to arbitrary data.
+
+#### 2.3 Tapscript
+
+[![Tapscript](/img/posts/taproot-workshop/tapscript.png)](https://www.youtube.com/watch?v=nXGe9_M5pjk&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=8)
+{:.center}
+
+James shows how using a taptweak, we can commit a tapscript in a segwit v1
+output, and how we can spend that output using the segwit v1 key path
+spend rules. He also explains the differences between tapscript and legacy
+bitcoin script.
+
+#### 2.4 Taptree
+
+[![Taptree](/img/posts/taproot-workshop/taptree.png)](https://www.youtube.com/watch?v=n6R15Eo6J44&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=9)
+{:.center}
+
+James shows how a merkle tree of scripts can be constructed, and how we can
+commit that tree using a taptweak. He then shows how to spend the output by
+satisfying one of those scripts and providing a proof that the script was part
+of the committed tree.
+
+#### 2.5 Huffman Construction
+
+_(No video)_ This bonus chapter shows how to most efficiently construct a tree
+of scripts, by placing scripts that are more likely to be spent closer to the
+root of the tree.
+
+## Case studies
+
+This section gives demonstrations of how the new schnorr/taproot technologies
+can be used to build advanced Bitcoin services and products.
+
+#### 3.1 Degrading Multisig
+
+_(No video)_ This chapter demonstrates a degrading multisig wallet. In all
+cases, the output can be spent with a subset of the 'live' keys, but after some
+timeout, the output can be spent with a mixture of 'live' and 'backup' keys.
+Taproot allows multiple spending paths to be committed to, and only the one
+that is exercised is revealed on chain.
+
+## Summary
+
+[![Summary](/img/posts/taproot-workshop/summary.png)](https://www.youtube.com/watch?v=Q1od076K7IM&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=10)
+{:.center}
+
+ John concludes the workshop by explaining how you can get involved in the
+community feedback process for these proposals.
+
+[slides]: /img/posts/taproot-workshop/taproot-workshop.pdf
+[transcript]: http://diyhpl.us/wiki/transcripts/bitcoinops/schnorr-taproot-workshop-2019/notes/
```

Comment:
https on this link please and thank you

Reply:
FYI, you could probably setup an automatically http→https redirect on your server.  The reason I (and it seems others) keep using http is because that's the URL we copied/pasted while viewing the transcript.